Rules/UseLiteralInitializerForHashtable.cs

```diff
@@ -0,0 +1,401 @@
+﻿// Copyright (c) Microsoft Corporation.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation.Language;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// A class to walk an AST to check if hashtable is not initialized using [hashtable]::new or new-object hashtable
+    /// </summary>
+    #if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    class UseLiteralInitializerForHashtable : AstVisitor, IScriptRule
+    {
+        private List<DiagnosticRecord> diagnosticRecords;
+        private HashSet<string> presetTypeNameSet;
+        private string fileName;
+
+        public UseLiteralInitializerForHashtable()
+        {
+            var presetTypeNames = new string[]
+            {
+                "system.collection.hashtable",
+                "hashtable"
```

Comment:
It's somewhat common to skip `system` namespace as it's implied, so I'd add `collection.hashtable`. I'd also use the correct case for each entry because people typically use the correct case in scripts and could help speed up the comparison some.